cordformation/src/test/resources/net/corda/plugins/DeployCordappWithDocker.gradle

```diff
@@ -0,0 +1,81 @@
+buildscript {
+    ext {
+        corda_group = 'net.corda'
+        corda_release_version = '4.0'
+        jolokia_version = '1.6.0'
+    }
+}
+
+plugins {
+    id 'net.corda.plugins.cordformation'
+}
+
+repositories {
+    mavenCentral()
+    maven { url 'https://software.r3.com/artifactory/corda-releases' }
+}
+
+dependencies {
+    runtime "$corda_group:corda:$corda_release_version"
+    runtime "$corda_group:corda-node-api:$corda_release_version"
+    cordapp "$corda_group:corda-finance-contracts:$corda_release_version"
+    cordapp "$corda_group:corda-finance-workflows:$corda_release_version"
+}
+
+task deployNodes(type: net.corda.plugins.Cordform) {
+    node {
+        projectCordapp {
+            deploy false
+        }
+        name 'O=Notary Service,L=Zurich,C=CH'
+        notary = [validating : true]
+        p2pPort 10002
+        rpcPort 10003
+        rpcSettings {
+            adminAddress "localhost:10004"
+        }
+        cordapps = ["$corda_group:corda-finance-contracts:$corda_release_version",
+                    "$corda_group:corda-finance-workflows:$corda_release_version"]
+    }
+}
+
+task prepareDockerNodes(type: net.corda.plugins.Dockerform, dependsOn: ['jar']) {
```

Comment:
One way around the OS/ENT harmonization issue is to potentially add a bit of configuration to this which is the `externalDatabasePort`. If it's set, use it as the port to a postgres instance, otherwise fall back to H2. We'd then have to document clearly that it won't work in OS because external DBs aren't supported there (again, assuming no harmonization). It might be worth doing this anyway as a work around for port clashes.

Reply:
@JamesHR3 If an external Postgres instance exists, then why not use the Corform instead of Dockerform?